File: RickAndMorty.Storage.Dtos/Planet.cs
namespace RickAndMorty.Storage.Dtos
{
    public class Planet
    {
        public int Id { get; set; }

        public required DateTime DateModified { get; set; }

        public required bool IsDeleted { get; set; }

        public required string Name { get; set; }

        public string? Type { get; set; }

        public string? Dimension { get; set; }
    }
}

File: RickAndMorty.WebUi/src/models/planet.ts
export class Planet {
    id: number;
    name: string | null;
    type: string | null;
    dimension: string | null;

    constructor()
    constructor(
        id: number,
        name: string | null,
        type: string | null,
        dimension: string | null
    );
    constructor(
        id?: number,
        name?: string | null,
        type?: string | null,
        dimension?: string | null
    ) {
        this.id = id ?? 0;
        this.name = name ?? null;
        this.type = type ?? null;
        this.dimension = dimension ?? null;
    }
}